DICOMPlugins/DICOMM3DPlugin.py

```diff
@@ -0,0 +1,641 @@
+      referencedName = self.defaultSeriesNodeName(loadable.referencedSeriesUID)
+    return referencedName
+
+  def getEncapsulatedDocument(self, candidateFile):
+    dcm = pydicom.read_file(candidateFile)
+    if hasattr(dcm, "EncapsulatedDocument"):
+      return dcm.EncapsulatedDocument
+    else:
+      return b''
+    
+  def getEncapsulatedDocumentLength(self, candidateFile):
+    dcm = pydicom.read_file(candidateFile)
+    if hasattr(dcm, "EncapsulatedDocumentLength"):
+      return dcm.EncapsulatedDocumentLength
+    else:
+      return 0
+
+  def load(self,loadable):
+    """ Load the DICOM M3D object
+    """
+    logging.debug('DICOM M3D load()')
+    try:
+      uid = loadable.uid
+      logging.debug('in load(): uid = '+uid)
+    except AttributeError:
+      return False
+
+    self.tempDir = slicer.util.tempDirectory()
+    print(self.tempDir)
+    try:
+        os.makedirs(self.tempDir)
+    except OSError:
+      pass
+    
+    stlFileName = slicer.dicomDatabase.fileForInstance(uid)
+    if stlFileName is None:
+      logging.error('Failed to get the filename from the DICOM database for ' + uid)
+      self.cleanup()
+      return False
+    
+    # Read Encapsulated Document attribute into binary IO buffer
+    read_buffer = io.BytesIO(loadable.doc)
+    
+    # Take care of trailing padding in Encapsulated Document
+    # If EncapsulatedDocumentLength is odd, 
+    # the EncapsulatedDocument was padded to even length, otherwise it was already of even length. 
+    buffer_view = read_buffer.getbuffer()
+    if (int(loadable.docLength) % 2) != 0:
+      buffer_view = buffer_view[0:(read_buffer.getbuffer().nbytes -1)]
+    
+    # Save processed binary IO buffer into a temporary .STL file
+    stlFilePath = os.path.join(self.tempDir, "temp.STL")
+    with open(stlFilePath, 'wb') as file: 
+      shutil.copyfileobj(read_buffer, file)
+    assert os.path.exists(stlFilePath)
+    file.close()
+
+    # Create Segmentation Node based on .STL file
+    self._createSegmentationNode(loadable, stlFilePath)
+    
+    self.cleanup()
+    
+    return True
+
+  def _createSegmentationNode(self, loadable, stlFilePath):
+    #create modelNode
+    modelNode = slicer.util.loadModel(stlFilePath)
+
+    # Initialize Segmentation Node
+    segmentationNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLSegmentationNode")
+    segmentationNode.SetName(loadable.name)
+
+    # Initialize Segmentation Display Node
+    segmentationDisplayNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLSegmentationDisplayNode")
+    segmentationNode.SetAndObserveDisplayNodeID(segmentationDisplayNode.GetID())
+
+    #Setup converters for further export to labelmap or model representations 
+    vtkSegConverter = vtkSegmentationCore.vtkSegmentationConverter
+    segmentation = vtkSegmentationCore.vtkSegmentation()
+    segmentation.SetMasterRepresentationName(vtkSegConverter.GetSegmentationBinaryLabelmapRepresentationName())
```

Comment:
Wouldn't it make more sense for the closed surface model to be the Master representation here, since that's what came in the file.  Converting to a labelmap first is lossy since it relies on a heuristic to determine the labelmap resolution.